Tree/check_for_bst.cpp

```diff
@@ -118,28 +118,25 @@ struct Node {
 //This was functional problem so the solution is as follows:
 
 bool isBST(Node* root) {
-    return checkkBST(root, INT_MIN, INT_MAX); //chekkBST will return true or false depending on whether the tree is BST or not
+    static node *prev = NULL;  
+      
+    // traverse the tree in inorder fashion  
+    // and keep track of prev node  
+    if (root)  
+    {  
+        if (!isBST(root->left))  
+        return false;  
+  
+        // Allows only distinct valued nodes  
+        if (prev != NULL &&  
+            root->data <= prev->data)  
+        return false;  
+  
+        prev = root;  
+  
+        return isBST(root->right);  
+    }  
+  
+    return true;  
```

Comment:
This code does not check grandparent or grandchildren nodes when checking for bst property. The implementation which is already there is correct.